### PR TITLE
go-owasm: Cleanup code documentation and error messages

### PR DESCRIPTION
**Pull request by sorawit**
- Add doc comments for Span and VMLogic
- Remove `GoResult` in favor of only use `Error` enum
- Better error enum names and explanation

### CODE REVIEW

File: chain/x/oracle/types/exec_env.go [hunk 1]
```diff
@@ -69,10 +69,10 @@ func NewPrepareEnv(req Request, maxRawRequests int64) *PrepareEnv {
 // AskExternalData implements Owasm ExecEnv interface.
 func (env *PrepareEnv) AskExternalData(eid int64, did int64, data []byte) error {
 	if int64(len(data)) > MaxRawRequestDataSize {
-		return api.ErrSpanExceededCapacity
+		return api.ErrSpanTooSmall
 	}
 	if int64(len(env.rawRequests)) >= env.maxRawRequests {
-		return api.ErrAskExternalDataExceed
+		return api.ErrWrongPeriodAction
```
Comment (taobun) [inline chain/x/oracle/types/exec_env.go]: I think this error doesn't com from wrong period

Comment (taobun) [reply chain/x/oracle/types/exec_env.go]: ErrTooManyExternalData ?

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/x/oracle/types/exec_env.go (hunk 1 above)
Comment (sorawit) [reply chain/x/oracle/types/exec_env.go]: Good catch

**Review by Benzbeeb — APPROVED** [review]

File: go-owasm/src/span.rs [hunk 2]
```diff
@@ -18,27 +26,24 @@ impl Span {
         }
     }
 
+    /// Creates a writable `span` with the provided `ptr` as the starting memory location,
+    /// and inital capacity `cap`. The created span has zero length.
```
Comment (perimeko) [inline go-owasm/src/span.rs]: typo `initial`

**Review by perimeko — APPROVED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**